src/Microsoft.Sbom.Api/PackageDetails/ComponentDetailsUtils/MavenUtils.cs

```diff
@@ -0,0 +1,187 @@
+        {
+            var componentNamespaceParts = componentNamespace.Split('.'); // Example: "org.apache.commons.commons-lang3"
+
+            for (var i = 0; i < componentNamespaceParts.Length; i++)
+            {
+                pomLocation += $"/{componentNamespaceParts[i]}";
+            }
+        }
+
+        pomLocation += $"/{componentName}/{componentVersion}/{componentName}-{componentVersion}.pom";
+
+        // Check for file permissions on the .m2 directory before attempting to check if the file exists
+        if (MavenPackagesPathHasReadPermissions)
+        {
+            if (fileSystemUtils.FileExists(pomLocation))
+            {
+                return pomLocation;
+            }
+            else
+            {
+                log.Debug($"Pom location could not be found at: {pomLocation}");
+            }
+        }
+
+        return null;
+    }
+
+    public (string Name, string Version, PackageDetails packageDetails) ParsePom(string pomLocation)
+    {
+        var supplierField = string.Empty;
+        var licenseField = string.Empty;
+
+        try
+        {
+            var pomBytes = fileSystemUtils.ReadAllBytes(pomLocation);
+            using var pomStream = new MemoryStream(pomBytes, false);
+
+            var doc = new XmlDocument();
+            doc.Load(pomStream);
+
+            XmlNode developersNode = doc["project"]?["developers"];
+            XmlNode licensesNode = doc["project"]?["licenses"];
+            XmlNode organizationNode = doc["project"]?["organization"];
+
+            var name = doc["project"]?["artifactId"]?.InnerText;
+            var version = doc["project"]?["version"]?.InnerText;
+
+            if (organizationNode != null)
+            {
+                var organizationName = organizationNode["name"]?.InnerText;
+                if (!string.IsNullOrEmpty(organizationName))
+                {
+                    supplierField = $"Organization: {organizationName}";
+                }
+            }
+            else if (developersNode != null)
+            {
+                // Take the first developer name and use it as the supplier when there is no organization listed.
+                var developerName = developersNode["developer"]?["name"]?.InnerText;
+                if (!string.IsNullOrEmpty(developerName))
+                {
+                    supplierField = $"Person: {developerName}";
+                }
+            }
+
+            if (licensesNode != null)
+            {
+                foreach (XmlNode licenseNode in licensesNode.ChildNodes)
+                {
+                    var licenseName = licenseNode["name"]?.InnerText;
+
+                    if (!string.IsNullOrEmpty(licenseName))
+                    {
+                        licenseField = licenseName;
+                    }
+                }
+            }
+
+            return (name, version, new PackageDetails(licenseField, supplierField));
+        }
+        catch (Exception e)
```

Comment:
Generally when you catch `Exception` instead of something more specific it's good to either justify in a comment why you can't allow failure here or catch something more specific.